### PR TITLE
Make use of featureUtility for minimum sized kernel images

### PR DESCRIPTION
**Pull request by leochr**
- Start with kernel binary and grow the image as necessary using _featureUtility_
- Introduce new tag `kernel-slim`
 -- `kernel` will remain unchanged to not break existing users, but will be deprecated and no longer be updated)
- Remove some configuration snippets that are not really useful (related to HTTP, JMS, IIOP)
- Split build configurations into 2 scripts. This would allow users to invoke the corresponding script at various stages and take advantage of cache:
-- **features.sh** : adds feature-related configuration snippets and use _featureUtility_ to grow the image as necessary for `defaultServer` (takes the user-specified server configuration into account as well)
-- **configure.sh** : adds server configuration for session cache, SSO, Keystore (when the corresponding flag is enabled) along with iFix installation and populating SCC.

Signed-off-by: leochr <leojc@ca.ibm.com>

### CODE REVIEW

**Comment by leochr** [comment]
@NottyCode @arthurdm Please review when you have a moment. Thank you.

**Comment by arthurdm** [comment]
thanks @leochr!  This is a great PR.  

Two comments:
* the `latest/full` Dockerfiles still reference 20.0.0.9 (should be at 20.0.0.10)
* can you please also make some updates to the main `readme.md`, to showcase the new patterns?

**Comment by leochr** [comment]
Thanks @arthurdm. I thought I rebased the latest release changes, but missed it for full tag. 

I'll create a separate PR for the documentation changes to include information about new tag, deprecation of `kernel` tag, deprecation of some enterprise functionality tags and example of new pattern and have it reviewed by ID.

**Review by arthurdm — APPROVED** [review]
LGTM - thanks @leochr!

**Comment by leochr** [comment]
@NottyCode please review when you get a chance. Thank you. We would like to include this with the 20.0.0.11 release. We are holding off the release for this PR.

File: releases/latest/full/Dockerfile.ubi.adoptopenjdk11 [hunk 1]
```diff
@@ -1,4 +1,79 @@
...
+    && chown -R 1001:0 /opt/ol/helpers \
+    && chmod -R g+rw /opt/ol/helpers \
+    && chown -R 1001:0 /opt/ol/fixes \
+    && chmod -R g+rwx /opt/ol/fixes \
+    && mkdir /etc/wlp \
+    && chown -R 1001:0 /etc/wlp \
+    && chmod -R g+rw /etc/wlp \
+    && echo "<server description=\"Default Server\"><httpEndpoint id=\"defaultHttpEndpoint\" host=\"*\" /></server>" > /config/configDropins/defaults/open-default-port.xml
 
 RUN cp /opt/ol/wlp/templates/servers/javaee8/server.xml /config/server.xml
```
Comment (NottyCode) [inline releases/latest/full/Dockerfile.ubi.adoptopenjdk11:78]: I think this line could be removed if the server create on line 47 specifies --template=javaee8

Comment (leochr) [reply releases/latest/full/Dockerfile.ubi.adoptopenjdk11]: Good point. Made changes to all Dockerfile for full tag. Thanks.

**Comment by leochr** [comment]
@NottyCode Addressed the review comment you added. Please let me know if you have any other comments/questions. Thank you.

**Comment by leochr** [comment]
Merging this PR now and will address any comments via a separate PR. Agreed by @NottyCode.

@crpotter you can start the release process. The `latest` has the `kernel-slim` tag instead of `kernel`.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@
 1. **Daily Images**
     *  Available [here](https://hub.docker.com/r/openliberty/daily), these are daily images from the daily Open Liberty binaries.  The scripts used for this image can be found [here](https://github.com/OpenLiberty/ci.docker.daily).
 
+_Note:_ `kernel` tag is now deprecated and it will not be updated (starting with 20.0.0.11). The new kernel tag is named `kernel-slim`.
```
Comment (NottyCode) [inline README.md:28]: we should update for 20.0.0.12.

**Pull request merged**